Post typescript improvements

williamlardier opened the pull request:
-> The validator does not make sense here as we use the function to also remove the bucket policy. The objective here is to be backward compatible.

Issue: ARSN-482

Seragonia committed:
Do not validate bucket policy in setter

The setter is also used to remove polcicies from the bucket

Issue: ARSN-482

Seragonia committed:
Bump arsenal version

Issue: ARSN-482

bert-e commented:
# Hello williamlardier,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/arsenal/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ARSN-482 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.2.10`


Please check the `Fix Version/s` of ARSN-482, or the target
branch of this pull request.

Seragonia committed:
Return routing rule instances in getRoutingRules

But return JSON object in getConfig.
Also, the routing rule is not mandatory in AWS, so default
to undefined.

Issue: ARSN-482

Seragonia committed:
Adapt externalBackendCopy with bucketMD defaults update

Issue: ARSN-482

ghost commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ARSN-482 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.2.10`


Please check the `Fix Version/s` of ARSN-482, or the target
branch of this pull request.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.1`



Please check the status of the associated issue ARSN-482.

Goodbye williamlardier.




*The following options are set:* **approve**
